framework/src/media/stream_info.c

```diff
@@ -0,0 +1,69 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <pthread.h>
+#include <stdbool.h>
+#include <media/stream_info.h>
+
+static pthread_mutex_t stream_info_mutex;
+static uint64_t stream_info_id;
+bool is_stream_info_init;
+
+void stream_info_init(void)
+{
+	if (is_stream_info_init) {
+		return;
+	}
+
+	pthread_mutex_init(&stream_info_mutex, NULL);
+	is_stream_info_init = true;
+}
+
+static uint64_t stream_info_id_generate(void)
+{
+	uint64_t id;
+	pthread_mutex_lock(&stream_info_mutex);
+	id = ++stream_info_id;
```

Comment:
If we maintain id with sequence number, then index is correct.

Reply:
We are using uint64_t as identifier. So I think id is also ok.
If we have better solution to make id, we can just modify this function.
At that time, we can't use the name(index).